server/pkcs11.go

```diff
@@ -0,0 +1,158 @@
+package server
+
+import (
+	"crypto"
+	"strconv"
+	"strings"
+	"net/url"
+	"regexp"
+
+	"github.com/thalesignite/crypto11"
+	"github.com/cloudflare/cfssl/log"
+)
+
+// PKCS11URI defines a PKCS#11 URI: https://tools.ietf.org/html/rfc7512#section-2.3
+type PKCS11URI struct { // pkcs11:path-attr[?query-attr]
+	// path-attr: (; delimited)
+	Token  string //        token <- CK_TOKEN_INFO
+	Manuf  string // manufacturer <- CK_TOKEN_INFO
```

Comment:
I believe the idiomatic Go thing would be to spell these fields out, i.e. `Manufacturer` instead of `Manuf`, `LibraryDescription` instead of `LibDesc`, etc.

Reply:
I changed `Id` to `ID` but as for others, I'd rather stick to the RFC convention here.